server/src/core/auth/strategies/jwt.auth.strategy.ts

```diff
@@ -24,6 +24,18 @@ export class JwtAuthStrategy extends PassportStrategy(Strategy, 'jwt') {
   }
 
   async validate(payload: JwtPayload): Promise<PassportUser> {
+    const workspaceFromJWT =
+      await this.prismaService.client.workspace.findUnique({
+        where: { id: payload.sub },
+      });
+    if (workspaceFromJWT) {
+      // If apiKey has been deleted, we throw an error
+      await this.prismaService.client.apiKey.findUniqueOrThrow({
+        where: { id: payload.jti },
+      });
+      return { user: undefined, workspace: workspaceFromJWT };
+    }
+
```

Comment:
@martmull Instead of adding this extra code, we can simply do:

```typescript
    const workspace =
      await this.prismaService.client.workspace.findUniqueOrThrow({
        where: { id: payload.workspaceId ?? payload.sub},
      });
    if (!workspace) {
      throw new UnauthorizedException();
    }
    // You need to check if the token is revoked !
    if (payload.jti) {
       const apiKey = await this.prismaService.client.apiKey.findUniqueOrThrow({
         where: { id: payload.jti },
       });
       
       assert(apiKey.revokedAt, "This API Key is revoked", ForbiddenException);
    }
    const user = payload.sub ? (await this.prismaService.client.user.findUniqueOrThrow({
       where: { id: payload.sub },
     })) : undefined;
    return { user, workspace };
```

Reply:
ok, lets use revokedAt, do I keep the deleteApiKey endpoint then?